### PR TITLE
chore: Fix validation messages for label title

### LINKED ISSUE #1298: enhancement: Add validation error for label create modal
# Enhancement 

Label names in chatwoot only support alphabets, numbers, hyphen and underscore.  
At the moment the available validation doesn't show the user any errors, which would leave the user confused. 

So let's add a validation error describing this behaviour

### Current behaviour
Screenshot 2020-10-02 at 4 29 03 PM

### Expected

Screenshot 2020-10-02 at 4 26 55 PM

### PR DESCRIPTION
**Pull request by m4tr1k**
# Pull Request Template

## Description

@sojan-official wanted to put error messages on label titles to notify the user what caused the following error (which is blank spaces). As soon as the label has space, the app notifies the user with an error message.

Fixes #1298 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

I locally tested this on my machine and it is showing the error message as soon as the label has space.
I also added and fixed the tests of the labels, to support this new functionality

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/assets/scss/widgets/_forms.scss [hunk 1]
```diff
@@ -32,3 +32,8 @@ input {
 .help-text {
   font-weight: $font-weight-normal;
 }
+
+.error-text {
+  color: darken(get-color(alert), 25%);
```
Comment (pranavrajs) [inline app/javascript/dashboard/assets/scss/widgets/_forms.scss:37]: Please use color variables from `dashboard/assets/scss/variables.scss` or `shareda/assets/stylesheets/colors.scss`

File: app/javascript/dashboard/assets/scss/widgets/_forms.scss [hunk 2]
```diff
@@ -32,3 +32,8 @@ input {
...
+  margin-bottom: 1.6rem !important;
```
Comment (pranavrajs) [inline app/javascript/dashboard/assets/scss/widgets/_forms.scss:38]: Please use the space variable. `(dashboard/assets/scss/variables.scss)`

File: app/javascript/dashboard/routes/dashboard/settings/labels/AddLabel.vue [hunk 3]
```diff
@@ -12,6 +12,8 @@
           class="medium-12 columns"
           :label="$t('LABEL_MGMT.FORM.NAME.LABEL')"
           :placeholder="$t('LABEL_MGMT.FORM.NAME.PLACEHOLDER')"
+          :errorText="'Only Alphabets, Numbers, Hyphen and Underscore are allowed!'"
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/labels/AddLabel.vue:15]: Please move this i18n file.

File: app/javascript/dashboard/routes/dashboard/settings/labels/validations.js [hunk 4]
```diff
@@ -1,12 +1,14 @@
 import { required, minLength } from 'vuelidate/lib/validators';
 
-export const validLabelCharacters = (str = '') => !!str && !str.includes(' ');
+export const validLabelCharacters = (str = '') => !str.includes(' ');
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/labels/validations.js:3]: Can we regex here instead of this?

**Review by pranavrajs — CHANGES_REQUESTED** [review]
@m4tr1k Thanks for the PR. I've some comments about the PR. Please take a look and let me know your thoughts.

**Comment by m4tr1k** [comment]
@pranavrajs of course, thanks for pointing those details. Gonna work on them soon and then you can review again 😄

**Comment by m4tr1k** [comment]
@pranavrajs let me know if there's anything to fix

**Comment by pranavrajs** [comment]
@m4tr1k Thanks for the fixes. I will test it out and see the behaviour and will let you know.

File: app/javascript/dashboard/assets/scss/widgets/_forms.scss [hunk 5]
```diff
@@ -32,3 +32,8 @@ input {
 .help-text {
   font-weight: $font-weight-normal;
 }
+
+.error-text {
+  color: $alert-color;
+  margin-bottom: $space-normal !important;
```
Comment (nithindavid) [inline app/javascript/dashboard/assets/scss/widgets/_forms.scss:38]: Is important mandatory here ?

File: app/javascript/dashboard/routes/dashboard/settings/labels/validations.js [hunk 6]
```diff
@@ -1,12 +1,14 @@
 import { required, minLength } from 'vuelidate/lib/validators';
 
-export const validLabelCharacters = (str = '') => !!str && !str.includes(' ');
+export const validLabelCharacters = (str = '') => !/ /.test(str);
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/labels/validations.js:3]: 👍

**Review by nithindavid — CHANGES_REQUESTED** [review]

**Comment by sojan-official** [comment]
Closing this PR since it hasn't been active for a while.

**Pull request closed without merging**